docs/introduction-to-smart-contracts.rst

```diff
@@ -396,26 +396,33 @@ returns that code when executed.
 Gas
 ===
 
-Upon creation, each transaction is charged with a certain amount of **gas**.
-Imposing a cost on transactions serves to secure the network:
-it deincentivizes bad actors from spamming, and pays validators (i.e. miners and stakers)
-for the work that is required to validate new blocks. While the EVM executes the
+Upon creation, each transaction is charged with a certain amount of **gas**
+that has to be paid for by the originator of the transaction (``tx.origin``).
+While the EVM executes the
 transaction, the gas is gradually depleted according to specific rules.
+If the gas is used up at any point (i.e. it would be negative),
+an out-of-gas exception is triggered, which ends execution and reverts all modifications
+made to the state in the current call frame. 
```

Comment:
I wanted to comment that it's not just the current call frame but the whole transaction since you cannot silence out-of-gas. Then I realized the caller could forward less gas and handle the revert. Maybe it would be a good idea to mention that the caller can decide how much gas he forwards?